### PR TITLE
First draft of rewriting library on top of opentracing

### PR DESCRIPTION
**Pull request by whiskeysierra**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: tracer-core/src/main/java/org/zalando/tracer/MDCSpanObserver.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import static org.zalando.tracer.Flow.Logging.TRACE_ID;
+
+/**
+ * %equals{%X{flow_id}}{}{%X{trace_id}}
+ */
+public final class MDCSpanObserver implements TracerObserver, SpanObserver {
+
+    @Override
+    public SpanObserver onStart(final SpanData data) {
+        // TODO https://github.com/opentracing-contrib/java-api-extensions/issues/17
```
Comment (whiskeysierra) [inline tracer-core/src/main/java/org/zalando/tracer/MDCSpanObserver.java]: This is kind of a blocker.

Comment (whiskeysierra) [reply tracer-core/src/main/java/org/zalando/tracer/MDCSpanObserver.java]: https://github.com/opentracing-contrib/java-api-extensions/issues/17

Comment (lukasniemeier-zalando) [reply tracer-core/src/main/java/org/zalando/tracer/MDCSpanObserver.java]: via https://github.com/opentracing/opentracing-java/issues/314

**Comment by whiskeysierra** [comment]
Could be worth adding a link to https://github.com/opentracing/specification/blob/master/rfc/trace_identifiers.md#log-correlation.

File: tracer-core/src/main/java/org/zalando/tracer/MDCSpanObserver.java (hunk 1 above)
Comment (whiskeysierra) [reply tracer-core/src/main/java/org/zalando/tracer/MDCSpanObserver.java]: First one is done

File: tracer-spring-boot-starter/src/main/java/org/zalando/tracer/spring/TracerAutoConfiguration.java [hunk 2]
```diff
@@ -208,18 +64,19 @@ public TaskExecutor taskExecutor(final ExecutorService taskExecutorService, fina
...
-            final Filter filter = new TracerFilter(tracer);
+        public FilterRegistrationBean flowFilter(final Flow flow) {
+            final Filter filter = new FlowFilter(flow);
             @SuppressWarnings("unchecked") // as of Spring Boot 2.x
             final FilterRegistrationBean registration = new FilterRegistrationBean(filter);
             registration.setName(FILTER_NAME);
-            registration.setDispatcherTypes(REQUEST, ASYNC);
-            registration.setOrder(Ordered.HIGHEST_PRECEDENCE);
+            registration.setDispatcherTypes(REQUEST);
+            // see https://github.com/opentracing-contrib/java-spring-web/blob/master/opentracing-spring-web-starter/src/main/java/io/opentracing/contrib/spring/web/starter/WebTracingProperties.java#L22
```
Comment (lukasniemeier-zalando) [inline tracer-spring-boot-starter/src/main/java/org/zalando/tracer/spring/TracerAutoConfiguration.java]: by now it's line `35`, hence I'd just omit the line number

**Pull request merged**